pkg/reconciler/kubernetes/tektonhub/tektonhub.go

```diff
@@ -0,0 +1,595 @@
+		err := r.applyManifest(ctx, dbLocation, th, dbInstallerSet, version, "hub-db")
+		if err != nil {
+			return err
+		}
+	}
+
+	err = r.checkComponentStatus(ctx, th, dbInstallerSet)
+	if err != nil {
+		th.Status.MarkDbInstallerSetNotAvailable(err.Error())
+		r.enqueueAfter(th, 10*time.Second)
+		return err
+	}
+
+	th.Status.MarkDbInstallerSetAvailable()
+
+	// Check if the InstallerSet is available for DB-migration
+	exist, err = checkIfInstallerSetExist(ctx, r.operatorClientSet, version, th, dbMigrationInstallerSet)
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+		dbMigrationLocation := filepath.Join(hubDir, "db-migration")
+		th.Status.MarkDatabasebMigrationFailed("DB migration installerset not available")
+		err = r.applyManifest(ctx, dbMigrationLocation, th, dbMigrationInstallerSet, version, "hub-db-migration")
+		if err != nil {
+			return err
+		}
+	}
+
+	err = r.checkComponentStatus(ctx, th, dbMigrationInstallerSet)
+	if err != nil {
+		th.Status.MarkDatabasebMigrationFailed(err.Error())
+		r.enqueueAfter(th, 10*time.Second)
+		return err
+	}
+
+	th.Status.MarkDatabasebMigrationDone()
+
+	// Validate whether the secrets and configmap are created for API
+	if err := r.validateApiDependencies(ctx, th); err != nil {
+		th.Status.MarkApiDependencyMissing("api secrets not present")
+		r.enqueueAfter(th, 10*time.Second)
+		return err
+	}
+
+	th.Status.MarkApiDependenciesInstalled()
+
+	// Create the API route based on platform
+	if err := r.extension.PreReconcile(ctx, th); err != nil {
+		return err
+	}
+
+	th.Status.MarkPreReconcilerComplete()
+
+	exist, err = checkIfInstallerSetExist(ctx, r.operatorClientSet, version, th, apiInstallerSet)
+	if err != nil {
+		return err
+	}
+
+	if !exist {
+		th.Status.MarkApiInstallerSetNotAvailable("API installer set not available")
+		apiLocation := filepath.Join(hubDir, "api")
+		err := r.applyManifest(ctx, apiLocation, th, apiInstallerSet, version, "hub-api")
+		if err != nil {
+			return err
+		}
+	}
+
+	err = r.checkComponentStatus(ctx, th, apiInstallerSet)
+	if err != nil {
+		th.Status.MarkApiInstallerSetNotAvailable(err.Error())
+		r.enqueueAfter(th, 10*time.Second)
+		return err
+	}
+
+	th.Status.MarkApiInstallerSetAvailable()
+
+	// Create the API route based on platform
+	if err := r.extension.PostReconcile(ctx, th); err != nil {
+		return err
+	}
+
+	th.Status.MarkPostReconcilerComplete()
+
+	return nil
+}
+
+// Validate DB are present on the cluster. If DB secrets are present and all the keys don't
+// have values then update the remaining values with default values. If the DB secret
+// is not present then create a new secret with default values.
+func (r *Reconciler) validateOrCreateDBSecrets(ctx context.Context, th *v1alpha1.TektonHub) error {
+	logger := logging.FromContext(ctx)
+	dbKeys := []string{"POSTGRES_DB", "POSTGRES_USER", "POSTGRES_PASSWORD", "POSTGRES_PORT"}
+
+	th.Status.MarkDbDependencyInstalling("db secrets are being added into the namespace")
```

Comment:
this is an overkill, this will not be updated anyway in status before the end of this reconcile call. So we can get rid of this.